src/utils/types/DataDescriptor.ts

```diff
@@ -0,0 +1,12 @@
+
+export type DataDescriptor = {
```

Comment:
Please rename to DataDescriptorInfo or something along those lines as to prevent clashing with the DataDescriptor class